packages/lesswrong/components/users/Leaderboard.tsx

```diff
@@ -0,0 +1,317 @@
+    );
+
+  // We'll separate donors into those with a known numeric donation, and those over 5k with no exact data
+  const donors = donorData?.AirtableLeaderboards || [];
+  const donorsWithAmounts = donors.filter(
+    (entry) => typeof entry.leaderboardAmount === "number"
+  );
+  const donorsOver5kUnknown = donors.filter(
+    (entry) => typeof entry.leaderboardAmount !== "number"
+  );
+
+  const last30DaysResults =
+    last30DaysData?.NetKarmaChangesForAuthorsOverPeriod || [];
+
+  // Add type guard functions
+  const isAllTimeEntry = (entry: UsersMinimumInfo | KarmaEntry): entry is UsersMinimumInfo => {
+    return '_id' in entry;
+  };
+
+  const isKarmaEntry = (entry: UsersMinimumInfo | KarmaEntry): entry is KarmaEntry => {
+    return 'userId' in entry;
+  };
+
+  return (
+    <ErrorBoundary>
+      <div className={classes.pageContainer}>
+        <h1 className={classes.header}>Leaderboard</h1>
+        <p className={classes.subTitle}>
+          Here are our top users and donors to <a href="https://lightconeinfrastructure.com">Lightcone Infrastructure</a>, the organization maintaining LessWrong. 
+        </p>
+
+        <div className={classes.columnsContainer}>
+          {/* Karma Leaderboard Column */}
+          <div className={classes.column}>
+            <div className={classes.headerRow}>
+              <h2 className={classes.header}>
+                {timeframe === "allTime" ? "Top Karma Users (All Time)" : "Top Karma Gains (30 Days)"}
+              </h2>
+              <div className={classes.toggleGroup}>
+                {["allTime", "30days"].map((period) => (
+                  <div
+                    key={period}
+                    className={classNames(classes.toggleButton, {
+                      [classes.activeToggle]: timeframe === period,
+                    })}
+                    onClick={() => setTimeframe(period as "allTime" | "30days")}
+                  >
+                    {period === "allTime" ? "All Time" : "30 Days"}
+                  </div>
+                ))}
+              </div>
+            </div>
+
+            <LeaderboardTable headers={["#", "User", "Karma"]}>
+              <tbody>
+                {(timeframe === "allTime" ? topKarmaUsers : last30DaysResults)?.map((entry, i) => (
+                  <tr key={isAllTimeEntry(entry) ? entry._id : entry.userId}>
+                    <td>{i + 1}</td>
+                    <td>
+                      {isAllTimeEntry(entry) 
+                        ? <UsersNameDisplay user={entry} />
+                        : <UsersName documentId={entry.userId} />
+                      }
+                    </td>
+                    <td>
+                      {(isAllTimeEntry(entry) ? entry.karma : entry.netKarma).toLocaleString()}
+                    </td>
+                  </tr>
+                ))}
+              </tbody>
+            </LeaderboardTable>
+            
+            <LoadMore {...(timeframe === "allTime" ? loadMoreProps : {
```

Comment:
I don't know if it's worth pulling out all of them, but I am generally not a huge fan of complicated ternaries inlined into components